### DISCUSSION TITLE
Field specified or default value in root validator

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [pydantic/pydantic#4646] [Question] [answered]
Can I figure out if a required field was manually given, or whether the default value is used?

I'd like to overwrite a value in a root validator if the default value is used, but not if the default value was manually given.

**Comment by samuelcolvin** [comment, marked as answer]
I think right now you'd need to do this after validation.

**Reply by WilliamDEdwards** [reply]
Am I not understanding your reply, or is your reply for something else?

**Reply by samuelcolvin** [reply]
Auto-correct, sorry. I've updated it.

**Reply by WilliamDEdwards** [reply]
By "after validation", do you mean after Pydantic has processed the model? If so, how would I know if the default value was passed manually or filled by Pydantic?

**Reply by samuelcolvin** [reply]
By looking at `__fields_set__`.
